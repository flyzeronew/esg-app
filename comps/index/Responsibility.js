import React, { useEffect, useState } from "react";
import classnames from "classnames/bind";
import styles from './Responsibility.module.css';

const cx = classnames.bind(styles);
const appUrl = process.env.APP_URL;

function Responsibility(props) {
    const [showBg, setShowBg] = useState(true);

    useEffect(() => {
        const checkWidth = () => {
            setShowBg(window.innerWidth > 1023);
        };
        checkWidth();
        window.addEventListener("resize", checkWidth);
        return () => window.removeEventListener("resize", checkWidth);
    }, []);

    const bgStyle = showBg
        ? { background: `url(${appUrl}/images/responsibility_bg01.jpg) no-repeat center center` }
        : {};

    return (
        <div className={cx("responsibility")} style={bgStyle}>
            <div className={cx("img")}>
                <img src={`${appUrl}/images/responsibility_bg01.jpg`} alt="responsibility_img01" width={768} height={641}/>
            </div>
            <div className={cx("box")}>
                <h2 className={cx("title")}>
                    Better Future, Together<br/>
                    永續共好
                </h2>
                <h1 className={cx("txt")}>
                    TVBS持續為閱聽眾帶來更全面，更淺顯易懂的永續體驗，共創關懷環境生態與新聞學的永續殿堂。
                </h1>
                <div className={cx("btn")}>
                    <a href={`${appUrl}/responsibility`}>
                        TVBS永續責任
                        <img src={`${appUrl}/images/icon_arraw_no_bg.svg`} alt="arrow_right" width={32} height={32}/>
                    </a>
                </div>
                <div className={cx("breadcrumbs")}>
                    <a href={`${appUrl}/responsibility/reports`}>
                        永續影響力報告
                        <img src={`${appUrl}/images/icon_arraw11.svg`} alt="arrow_right" width={12} height={12}/>
                    </a>
                    <a href={`${appUrl}/responsibility/guide`}>
                        永續e指南
                        <img src={`${appUrl}/images/icon_arraw11.svg`} alt="arrow_right" width={12} height={12}/>
                    </a>
                </div>
            </div>
        </div>
    );
}

export default Responsibility;
