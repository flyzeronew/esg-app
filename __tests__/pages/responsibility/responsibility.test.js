import { render,screen } from '@testing-library/react';
import Responsibility,  { getServerSideProps } from '@/pages/responsibility';
import { useRouter, useSearchParams, usePathname } from 'next/navigation';
global.fetch = jest.fn();

describe('responsibility page', () => {

  beforeEach(()=>{
    fetch.mockClear();
  })
  // const useRouter = jest.spyOn(require("next/navigation"), "usePathname");
  const mockPathName  = jest.spyOn(require("next/navigation"), 'usePathname')
                    .mockImplementation(() => mockPathname());
  describe("responsibility page test", ()=>{
    // useRouter.mockImplementation("/view");
    mockPathName.mockReturnValue('/view')
    const responsibilityProps = {
      menu:[
        {
          "page": "未來焦點",
          "pathname": "/focus",
          "title": "Focus TVBS",
          "meta_description": "Focus Meta description",
          "page_description": "Focus page description",
          "subMenu": []
        },
        {
            "page": "View Pages",
            "pathname": "/view",
            "page": "View page",
            "title": "View TVBS",
            "meta_description": "View Meta description",
            "page_description": "View page description",
            "subMenu": [
                {
                    "page": "Home View",
                    "pathname": "/",
                    "title": "Home view TVBS",
                    "meta_description": "View Meta description",
                    "page_description": "View page description",
                },
                {
                    "page": "Trending",
                    "pathname": "/trend",
                    "title": "Trending view TVBS",
                    "meta_description": "Trending Meta description",
                    "page_description": "Trending page description",
                },
                {
                    "page": "Environment",
                    "pathname": "/environment",
                    "title": "environment view TVBS",
                    "meta_description": "environment Meta description",
                    "page_description": "environment page description",
                }
    
            ]
        }
    ],

      responsibilityData:[{
          "article_id": 293,
          "cover_img": "https://esg-asset.tvbs.com.tw/articles/img/XU5hkGPrAtlKGokeUBjb1PiDGawTfxnf2qbk2rpk.jpg",
          "title": "地球解方關注全球塑膠公約，縮短價值行動差是永續溝通的最大挑戰",
          "shortTitle": "地球解方關注全球塑膠公約，縮短價值行動差",
          "url": "https://esg.tvbs.com.tw/view/trend/293",
          "description": "",
          "partner": {
              "id": 1,
              "name": "蝦皮商城",
              "logo": "/images/logo-view2.png",
              "introduction": "秉持「從心出發，實現美好生活」為核心，提供優質的產品與服務。"
          },
          "author_name": "flyzero",
          "article_genres": [
              {
                  "id": 4,
                  "name": "綠色生活",
                  "en_name": "greenlife",
              }
          ],
          "created_at": "2024-03-11T01:05:01.000000Z",
          "updated_at": "2024-03-11T01:05:01.000000Z"
      }],
      brands:[{
        id: 8,
        name: 'Super taste',
        brand_url: 'https://supertaste.tvbs.com.tw/',
        description: 'Food, restaurant and tourist attractions',
        cover_img: 'https://esg-statics-staging.tvbs.com.tw/brands/img/cover/UaR0U5duERc1zoaCuTkdfoeOFatadwy53b92HJYq.png',
        logo: 'https://esg-statics-staging.tvbs.com.tw/brands/img/logo/yVJrGQnA4DWZADVpO8eVotU4IIEOhZftli6IyMP5.png'
      }]
    }
    it("renders respsonsibility page unchanged", () => {
      const { container } = render(<Responsibility {...responsibilityProps}/>);
      // screen.debug();
      expect(container).toMatchSnapshot();
    });

    it("renders responsibility page with brand values",()=>{
      render(<Responsibility {...responsibilityProps} />);
      const brand = screen.getAllByRole('brandCard');
      expect(brand[0]).toBeInTheDocument();

    })

    // it("renders responsibility page without brand values",()=>{
    //   const responsibilityEmptyProps = {
    //     menu:[],
    //     responsibilityData:[],
    //     brands:[]
    //   }
    //   render(<Responsibility {...responsibilityEmptyProps} />);
    //   const brandsList = screen.getByRole('list', {name: 'brandCardsList'});
    //   expect(brandsList.children).toHaveLength(0)
    // })
  })
  
});
