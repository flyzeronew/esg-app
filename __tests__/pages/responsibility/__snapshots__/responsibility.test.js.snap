// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`responsibility page responsibility page test renders respsonsibility page unchanged 1`] = `
<div>
  <div
    id="wrapper"
  >
    <header
      class="header"
    >
      <div
        class="mainMenuBox"
      >
        <nav
          class="nav"
        >
          <div
            class="mainMenu"
          >
            <div
              class="logo"
            >
              <a
                href="/"
              >
                <img
                  alt="TVBS GOOD"
                  height="48"
                  src="/images/esg-logo.svg"
                  width="106.3"
                />
              </a>
            </div>
            <div
              class="moMediaSharing"
            >
              <div
                class="socialMedia"
              >
                <div>
                  <a
                    href="https://www.instagram.com/tvbs_good/"
                    target="_blank"
                  >
                    <img
                      alt="IG Desk Icon"
                      height="28"
                      src="/images/instagramIcon.svg"
                      width="28"
                    />
                  </a>
                </div>
                <div>
                  <a
                    href="https://www.youtube.com/@tvbsgood"
                    target="_blank"
                  >
                    <img
                      alt="IG Desk Icon"
                      height="28"
                      src="/images/youtubeIcon.svg"
                      width="28"
                    />
                  </a>
                </div>
              </div>
              <div
                class="burgerIcon"
              >
                <img
                  alt="img burger icon"
                  height="48"
                  loading="lazy"
                  src="/images/icon-burger.svg"
                  width="187"
                />
              </div>
            </div>
            <div
              class="object"
            >
              <div
                class="menu"
              >
                <ul>
                  <li>
                    <a
                      class=""
                      href="/focus"
                    >
                      未來焦點
                    </a>
                  </li>
                  <li>
                    <a
                      class=""
                      href="/view"
                    >
                      View page
                      <img
                        alt="arraw"
                        class=""
                        height="5"
                        src="/images/icon_arraw01.svg"
                        width="8"
                      />
                    </a>
                  </li>
                </ul>
              </div>
              <div
                class="socialMedia"
              >
                <div>
                  <a
                    href="https://www.instagram.com/tvbs_good/"
                    target="_blank"
                  >
                    <img
                      alt="IG Icon"
                      height="28"
                      src="/images/instagramIcon.svg"
                      width="28"
                    />
                  </a>
                </div>
                <div>
                  <a
                    href="https://www.youtube.com/@tvbsgood"
                    target="_blank"
                  >
                    <img
                      alt="Youtube Icon"
                      height="28"
                      src="/images/youtubeIcon.svg"
                      width="28"
                    />
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            class="childMenu"
          >
            <ul>
              <li>
                <a
                  href="/"
                >
                  Home View
                </a>
              </li>
              <li>
                <a
                  href="/trend"
                >
                  Trending
                </a>
              </li>
              <li>
                <a
                  href="/environment"
                >
                  Environment
                </a>
              </li>
            </ul>
          </div>
        </nav>
      </div>
    </header>
    <main>
      <div
        class="responsibilityPage"
      >
        <div
          class="mainView"
        >
          <div
            class="img"
          >
            <img
              alt="bg"
              height="1024"
              loading="lazy"
              src="https://esg-frontend-staging.tvbs.com.tw/images/responsibility_bg01.jpg"
              width="1920"
            />
          </div>
          <div
            class="titleBox"
          >
            <div
              class="title"
            >
              <div
                class="box"
              >
                <div
                  class="rounded"
                />
                <h1
                  class="t1"
                >
                  TVBS GOOD 永續共好
                </h1>
                <div
                  class="rounded"
                >
                  <img
                    alt="rounded"
                    height="50"
                    loading="lazy"
                    src="https://esg-frontend-staging.tvbs.com.tw/images/rounded-04.svg"
                    width="50"
                  />
                </div>
              </div>
              <div
                class="box"
              >
                <div
                  class="rounded"
                >
                  <img
                    alt="rounded"
                    height="50"
                    loading="lazy"
                    src="https://esg-frontend-staging.tvbs.com.tw/images/rounded-03.svg"
                    width="50"
                  />
                </div>
                <p>
                  Better Future.
                  <br />
                  Together
                </p>
                <div
                  class="rounded"
                >
                  <img
                    alt="rounded"
                    height="50"
                    loading="lazy"
                    src="https://esg-frontend-staging.tvbs.com.tw/images/rounded-04.svg"
                    width="50"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="content"
        >
          <div
            class="title"
          >
            <div
              class="line"
            />
          </div>
          <div
            class="mainImage"
          >
            <div
              class="img"
            >
              <div
                class="playIcon"
              >
                <img
                  alt="play"
                  height="50"
                  loading="lazy"
                  src="https://esg-frontend-staging.tvbs.com.tw/images/play-icon.svg"
                  width="50"
                />
              </div>
              <div
                class="txt"
              />
              <div
                class="video"
              >
                <img
                  alt="img"
                  height="1080"
                  loading="lazy"
                  src="/images/responsibility-video-img.png"
                  width="1920"
                />
              </div>
            </div>
          </div>
          <div
            class="contxt"
          >
            <div
              class="t1"
            >
              <div
                class="line"
              />
              <p>
                積極倡議永續 是我們堅定的承諾
              </p>
              <div
                class="line"
              />
            </div>
            <p>
              在永續議題上，沒有人是局外人。因為永續攸關人類發展，更是決定地球存亡的關鍵！作為媒體領導品牌，我們深刻理解媒體在塑造公眾意識和影響社會行為方面，擁有巨大的影響力。TVBS新聞長期製作深度專題報導，探討聯合國17項永續發展目標 (SDGs) 議題，包括監督環境危機的「帳篷下的秘密」，檢視長照、醫療制度的「你我老之路-老病死之生路」，關注派遣勞工權益的「派遣工的正義-我不是你的免洗筷」，以及探究人生老化過程中、各種照顧困境的「大照顧時代」，都透過深入淺出的剖析，傳達環境保護及永續理念，並獲得卓越新聞獎肯定。
            </p>
            <p>
              各內容品牌也都提出具體的ESG主張，例如:女人我最大節目，倡議「純淨美容Clean Beauty」的觀念，在變美之餘，也不造成地球環境負擔；地球黃金線關注電動車與環保材質汽車的最新趨勢；食尚玩家提倡綠色低碳旅遊等，皆是永續觀念的「具體轉譯」，與觀眾一起培養永續識讀力。唯有讓永續成為日常，才能創造「韌性」的地球。我們深切期盼各界攜手合作，共同探索解決氣候變遷、全球暖化和再生能源等重要議題的方法，並自我期許將成為媒體界的ESG貢獻者，為實現永續發展的美好願景盡心盡力。
            </p>
            <p
              style="text-align: right;"
            >
              聯利媒體 TVBS 總經理
              <br />
              永續發展委員會 主任委員  劉文硯
            </p>
          </div>
          <div
            class="reportArea"
          >
            <div
              class="reportLogo"
            >
              <img
                src="images/responsibility-report-logo.svg"
              />
            </div>
            <div
              class="reportTitle"
            >
              永續影響力報告
            </div>
            <div
              class="reportEnTitle"
            >
              看TVBS如何實踐永續承諾
            </div>
            <div
              class="downloadArea"
            >
              <a
                href="/responsibility/reports"
              >
                <div
                  class="desktop bar"
                >
                  <div
                    class="word"
                  >
                    了解更多
                  </div>
                  <div
                    class="arraw"
                  >
                    <img
                      alt="arraw"
                      height="36"
                      loading="lazy"
                      src="/images/icon_arraw02.svg"
                      width="36"
                    />
                  </div>
                </div>
              </a>
            </div>
          </div>
          <div
            class="infoGraphic"
          >
            <p>
              TVBS在新聞專題及各節目中，長期關注聯合國「2023 永續發展目標」 (Sustainable Development Goals, SDGs) 17 項核心目標的精神與價 值。為進一步善盡永續轉譯者、倡議者的角色，我們列出「四大倡議 願景」，並分別對應 7 項 SDG 作為重點，期發揮媒體影響力，與閱聽大 眾一起努力，邁向永續共好。
            </p>
          </div>
          <div
            class="block"
          >
            <div
              class="img"
            >
              <img
                alt="img"
                class="pc"
                loading="lazy"
                src="images/responsibility-block.svg"
              />
              <img
                alt="img"
                class="mo"
                loading="lazy"
                src="images/responsibility-block-mo.svg"
              />
            </div>
          </div>
          <div
            class="peopleBox"
          >
            <div
              class="title"
            >
              <h2>
                TVBS永續倡議大使 莊開文
              </h2>
              <div
                class="line"
              />
            </div>
            <div
              class="contxt"
            >
              <p>
                TVBS資深主播
                <strong>
                  莊開文
                </strong>
                ，擔任「TVBS永續倡議大使」，我們將從「永續、共好、愛無限」三面向積極推進ESG策略，與您攜手，共創美好的永續殿堂。
              </p>
            </div>
            <div
              class="people"
            >
              <div
                class="img"
              >
                <img
                  alt="img"
                  height="1080"
                  loading="lazy"
                  src="https://esg-frontend-staging.tvbs.com.tw/images/people01.jpg"
                  width="1920"
                />
              </div>
              <div
                class="txtBox"
              >
                <div
                  class="txt"
                >
                  我們深信，媒體的傳播力量可以撼動生命，影響一整個世代; 而您的關心和投入，更是這股力量的加速器!
                </div>
                <div
                  class="small"
                >
                  - 永續倡議大使 莊開文 
                </div>
              </div>
            </div>
            <div
              class="line"
            />
            <p>
              誠如 TVBS GOOD 的品牌識別設計，中間兩個 O 相連相握，便成為一個「無限大」的符號，這正是我們的核心精神：永續、共好、愛無限。面對氣候劇烈變遷、自然資源枯竭、糧食分配不均、疾病災害兵戎屢現等嚴酷考驗，我們需要更堅定的意志、更多元的學習、更包容的態度、更具體的作為。
            </p>
            <p>
              TVBS 在新聞專題及各節目中，長期關注聯合國「 2030 永續發展目標 」( Sustainable Development Goals, SDGs ) 17 項核心目標的精神與價值，為進一步善盡永續轉譯者、倡議者的角色，我們列出「四大倡議願景」，並分別對應 7 項 SDG 作為重點，期待在永續議題上，成為臺灣的推手，做世界的通道，為滋養我們的地球，盡心盡力。
            </p>
          </div>
          <div
            class="brand"
          >
            <div
              class="title"
            >
              <h2>
                TVBS永續品牌
              </h2>
              <div
                class="line"
              />
            </div>
            <p>
              根據英國牛津大學調查認證: TVBS連續7年獲選為最受信賴商業電視台，身為臺灣媒體領導品牌，我們秉持「真實」（Truth）、「信賴」（Trust）、「科技」（Technology）三大核心價值，發展出各大永續品牌。包括:倡議「純淨美容clean Beauty」觀念的「女人我最大」，關注電動車與環保材質汽車最新趨勢的「地球黃金線」，提倡綠色低碳旅遊的「食尚玩家」，關懷健康生活的「健康 2.0」，我們用觀眾能夠接受和理解的方式，將永續觀念「轉譯」為更親和易懂的內容。
            </p>
            <div
              class="listBox"
            >
              <div
                class="list"
              >
                <ul
                  aria-label="brandCardsList"
                >
                  <li
                    role="brandCard"
                  >
                    <a
                      href="https://esg-frontend-staging.tvbs.com.tw/responsibility/brands/8"
                    >
                      <div
                        class="imgBox"
                      >
                        <div
                          class="img"
                        >
                          <img
                            alt="img"
                            height="603"
                            loading="lazy"
                            src="https://esg-statics-staging.tvbs.com.tw/brands/img/cover/UaR0U5duERc1zoaCuTkdfoeOFatadwy53b92HJYq.png"
                            width="1072"
                          />
                        </div>
                        <div
                          class="logoBox"
                        >
                          <div
                            class="rounded"
                          >
                            <img
                              alt="rounded"
                              height="50"
                              loading="lazy"
                              src="https://esg-frontend-staging.tvbs.com.tw/images/rounded-04.svg"
                              width="50"
                            />
                          </div>
                          <div
                            class="logoFlex"
                          >
                            <div
                              class="logo"
                            >
                              <img
                                alt="logo"
                                layout="fill"
                                loading="lazy"
                                src="https://esg-statics-staging.tvbs.com.tw/brands/img/logo/yVJrGQnA4DWZADVpO8eVotU4IIEOhZftli6IyMP5.png"
                              />
                            </div>
                            <div
                              class="rounded"
                            >
                              <img
                                alt="rounded"
                                height="50"
                                loading="lazy"
                                src="https://esg-frontend-staging.tvbs.com.tw/images/rounded-04.svg"
                                width="50"
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="txt"
                      >
                        <div
                          class="description"
                        >
                          Food, restaurant and tourist attractions
                        </div>
                        <img
                          alt="rounded"
                          height="50"
                          loading="lazy"
                          src="https://esg-frontend-staging.tvbs.com.tw/images/icon_arraw09.svg"
                          width="50"
                        />
                      </div>
                    </a>
                  </li>
                </ul>
              </div>
            </div>
            <div
              class="title s1"
            >
              <h2>
                領航ESG媒體 實踐永續承諾
              </h2>
              <div
                class="line"
              />
            </div>
            <p>
              永續不是口號，而是關乎人類生存的關鍵字。
              <br />
              <br />
              我們希望透過永續領航者的角色，號召更多媒體、民眾能一起加入關注永續議題的行列，讓永續成為一門顯學。
              <br />
              <br />
              除了對台灣這片土地的關懷，TVBS未來也將跨足國際，發揮媒體影響力，期待在永續議題上，成為臺灣的推手，做世界的通道，為滋養我們的地球，盡心盡力。
            </p>
            <div
              class="carousel"
            >
              <div
                class="CustomSlider"
                role="slider"
              >
                <div
                  class="slick-slider slick-initialized"
                  dir="ltr"
                >
                  <div
                    class="slick-list"
                    style="padding: 0px 500px;"
                  >
                    <div
                      class="slick-track"
                      style="width: -1000px; opacity: 1; transform: translate3d(0px, 0px, 0px);"
                    >
                      <div
                        aria-hidden="false"
                        class="slick-slide slick-active slick-center slick-current"
                        data-index="0"
                        style="outline: none; width: -1000px;"
                        tabindex="-1"
                      >
                        <div>
                          <a
                            href="https://esg.tvbs.com.tw/view/trend/293"
                            style="width: 100%; display: inline-block;"
                            tabindex="-1"
                          >
                            <div
                              class="img"
                            >
                              <img
                                alt="img"
                                height="603"
                                src="https://esg-asset.tvbs.com.tw/articles/img/XU5hkGPrAtlKGokeUBjb1PiDGawTfxnf2qbk2rpk.jpg"
                                width="1072"
                              />
                            </div>
                            <div
                              class="txt"
                            >
                              <p
                                class="pc"
                              >
                                地球解方關注全球塑膠公約，縮短價值行動差是永續溝通的最大挑戰
                              </p>
                              <p
                                class="mo"
                              >
                                地球解方關注全球塑膠公約，縮短價值行動差
                              </p>
                            </div>
                          </a>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
    <div
      class="footerLine color1"
    >
      <div
        class="box"
      />
    </div>
    <footer
      class="footer"
    >
      © TVBS Media Inc. All Rights Reserved 
      <br />
      聯利媒體股份有限公司
    </footer>
  </div>
</div>
`;
